src/status_im/communities/core.cljs

```diff
@@ -76,14 +76,28 @@
       (update :chats <-chats-rpc)
       (update :categories <-categories-rpc)))
 
-(defn fetch-community-id-input
+(defn- fetch-community-id-input
   [{:keys [db]}]
   (:communities/community-id-input db))
 
+(defn- handle-my-request
+  [db {:keys [id community-id] :as request} state]
+  (if (= constants/community-request-to-join-state-pending state)
+    (assoc-in db [:communities/my-requests-to-join community-id id] request)
+    (update-in db [:communities/my-requests-to-join community-id] dissoc id)))
+
+(defn handle-admin-request
+  [db {:keys [id community-id] :as request}]
+  {:db (assoc-in db [:communities/requests-to-join community-id id] request)})
+
 (rf/defn handle-request-to-join
   [{:keys [db]} r]
-  (let [{:keys [id community-id] :as request} (<-request-to-join-community-rpc r)]
-    {:db (assoc-in db [:communities/requests-to-join community-id id] request)}))
+  (let [my-public-key                                          (get-in db [:multiaccount :public-key])
+        {:keys [id community-id state public-key] :as request} (<-request-to-join-community-rpc r)
+        is-my-request?                                         (= my-public-key public-key)]
+    (if is-my-request?
```

Comment:
It's probably old news already, but the `is-` prefix is redundant most of the time when there's a question mark. `my-request?` is just a bit more idiomatic.